blues/simulation.py

```diff
@@ -433,21 +445,75 @@ def simulateMD(self):
         self.nc_context.setPositions(md_state0['positions'])
         self.nc_context.setVelocities(md_state0['velocities'])
 
-    def run(self):
+    def runNCMC(self):
         """Function that runs the BLUES engine to iterate over the actions:
         Perform NCMC simulation, perform proposed move, accepts/rejects move,
         then performs the MD simulation from the NCMC state.
         """
         #set inital conditions
+        def _ncmcSanityCheck():
+            """Fuction that checks self attributes to see if they make sense for an NCMC simulation"""
+            if (self.nstepsNC % 2) != 0:
+                raise ValueError('nstepsNC needs to be even to ensure the protocol is symmetric (currently %i)' % (self.nstepsNC))
+            if self.nstepsNC <= 0:
+                raise ValueError('nstepsNC needs to be even and greater than zero so that perturbations can be relaxed via NCMC (currently nstepsNC=%i)' % (self.nstepsNC))
+
+        _ncmcSanityCheck()
         self.setStateConditions()
         for n in range(self.nIter):
             self.current_iter = int(n)
             self.setStateConditions()
             self.simulateNCMC(verbose=self.verbose, write_ncmc=self.write_ncmc)
-            self.acceptReject()
+            self.acceptRejectNCMC()
             self.simulateMD()
 
         # END OF NITER
         self.accept_ratio = self.accept/float(self.nIter)
         print('Acceptance Ratio', self.accept_ratio)
         print('nIter ', self.nIter)
+
+    def simulateMC(self):
+        """Function that performs the MC simulation."""
+
+        #choose a move to be performed according to move probabilities
+        self.move_engine.selectMove()
+        #change coordinates according to Moves in MoveEngine
+        new_context = self.move_engine.runEngine(self.md_sim.context)
+        md_state1 = self.getStateInfo(new_context, self.state_keys)
+        self.setSimState('md', 'state1', md_state1)
+
+    def acceptRejectMC(self):
+        """Function that chooses to accept or reject the proposed move.
+        """
+        md_state0 = self.current_state['md']['state0']
+        md_state1 = self.current_state['md']['state1']
+        log_ncmc = (md_state1['potential_energy'] - md_state0['potential_energy']) * (-1.0/self.nc_integrator.kT)
+        randnum =  math.log(np.random.random())
+
+        if log_ncmc > randnum:
+            self.accept += 1
+            print('NCMC MOVE ACCEPTED: log_ncmc {} > randnum {}'.format(log_ncmc, randnum) )
+            self.md_sim.context.setPositions(md_state1['positions'])
+        else:
+            self.reject += 1
+            print('NCMC MOVE REJECTED: log_ncmc {} < {}'.format(log_ncmc, randnum) )
```

Comment:
You've still got references to NCMC here in your MC stuff -- "NCMC move rejected", "log_ncmc", etc. Have some pride in your work -- make sure the MC part only uses "MC" not "NCMC". Make it look nice. :man_in_tuxedo: